torch/csrc/autograd/VariableTypeUtils.h

```diff
@@ -65,13 +65,13 @@ inline void rebase_history(Variable& var, std::shared_ptr<Function> grad_fn) {
   }
 }
 
-inline void rebase_history(ArrayRef<Variable> vars, std::shared_ptr<Function> grad_fn) {
+inline void rebase_history(std::vector<Variable>&& vars, std::shared_ptr<Function> grad_fn) {
   if (grad_fn) {
     for (auto& var : vars) {
       if (var.defined()) {
         // TODO: eliminate const_cast
         auto output_nr = grad_fn->add_input_metadata(var);
-        const_cast<Variable&>(var).rebase_history({grad_fn, output_nr});
```

Comment:
I think a better solution here is to declare `rebase_history` as a const method.

Reply:
No, it shouldn't be a `const` method. It mutates the state of the `Variable`